shell/browser/api/electron_api_web_contents.cc

```diff
@@ -1173,6 +1173,11 @@ void WebContents::DidStartLoading() {
 
 void WebContents::DidStopLoading() {
   Emit("did-stop-loading");
+
+  auto* web_preferences = WebContentsPreferences::From(web_contents());
+  if (web_preferences &&
+      web_preferences->IsEnabled(options::kPreferredSizeMode))
+    web_contents()->GetRenderViewHost()->EnablePreferredSizeMode();
```

Comment:
The location for calling this is based on Chromium's implementation: https://source.chromium.org/chromium/chromium/src/+/master:components/guest_view/browser/guest_view_base.cc;l=573;drc=d1252b1460e53a48e235c3bd9555bf5926c49100